src/common/utility.cpp

```diff
@@ -277,15 +279,18 @@ bool Utility::fsCasePreserving()
 
 bool Utility::fileNamesEqual(const QString &fn1, const QString &fn2)
 {
-    const QDir fd1(fn1);
-    const QDir fd2(fn2);
-
+    const QFileInfo fd1(fn1);
+    const QFileInfo fd2(fn2);
+    qDebug() << fd1.canonicalFilePath() << fd2.canonicalFilePath();
     // Attention: If the path does not exist, canonicalPath returns ""
     // ONLY use this function with existing pathes.
+    Q_ASSERT(fd1.exists());
+    Q_ASSERT(fd2.exists());
+    Q_ASSERT(fd1.canonicalFilePath() == fn1);
```

Comment:
You are calculating these canonical paths below anyway.

Reply:
ITS A WIP
and the idea is that I can remove the cannonical path one day

Reply:
I just created it to fix finish it one day